File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.py
class Solution:
    def minElement(self, nums: List[int]) -> int:
        
        def fun(a):
            su = 0 
            while a > 0:
                r = a % 10
                su = su + r
                a = a // 10
            
            return su

        res = []
        for i in range(len(nums)):
            res.append(fun(nums[i]))
        print(res)
        return min(res)

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.cpp
class Solution {
public:
    int sun(int n){
        int c=0;
        while(n!=0){
            int d=n%10;
            c+=d;
            n=n/10;
        }
        return c;
    }
    int minElement(vector<int>& nums) {
        for(int i=0;i<nums.size();i++){
            nums[i]=sun(nums[i]);
        }
        return *min_element(nums.begin(),nums.end());
    }
};